### PR TITLE
Add user-agent parameter to config

### PR DESCRIPTION
**Pull request by yu-kioo**
## overview
add optional user-agent parameter

### CODE REVIEW

**Review by NamedPython — APPROVED** [review]

File: src/main/java/org/embulk/input/DatabricksInputPlugin.java [hunk 1]
```diff
@@ -75,6 +90,13 @@ protected JdbcInputConnection newConnection(PluginTask task) throws SQLException
       props.put("ConnSchema", t.getSchemaName().get());
     }
     props.putAll(t.getOptions());
+    // overwrite UserAgentEntry property if the same property is set in options
+    if (t.getUserAgentEntry() != null) {
```
Comment (chikamura) [inline src/main/java/org/embulk/input/DatabricksInputPlugin.java:94]: `t.getUserAgentEntry` always seems to be not null.
You might want to delete this line.

**Review by chikamura — CHANGES_REQUESTED** [review]

File: src/main/java/org/embulk/input/DatabricksInputPlugin.java [hunk 2]
```diff
@@ -75,6 +90,13 @@ protected JdbcInputConnection newConnection(PluginTask task) throws SQLException
...
+      String product_name = t.getUserAgentEntry().getProductName();
+      String product_version = t.getUserAgentEntry().getProductVersion();
+
+      props.put("UserAgentEntry", product_name + "/" + product_version);
```
Comment (chikamura) [inline src/main/java/org/embulk/input/DatabricksInputPlugin.java:98]: Java's variable names are usually camel case.

product_name -> productName
product_version -> productVersion

File: src/main/java/org/embulk/input/DatabricksInputPlugin.java (hunk 1 above)
Comment (yu-kioo) [reply src/main/java/org/embulk/input/DatabricksInputPlugin.java]: The current code did not guarantee the behavior of 2 below, so changed condition.

- 1: options is not set & user-agent is not set => unknown/0.0.0
- 2: options is set & user-agent is not set => value of options
- 3: options is not set & user-agent is set => value of user-agent
- 4: options is set, user-agent is set => value of user-agent

File: src/main/java/org/embulk/input/DatabricksInputPlugin.java (hunk 2 above)
Comment (yu-kioo) [reply src/main/java/org/embulk/input/DatabricksInputPlugin.java]: sorry, fixed.

File: src/main/java/org/embulk/input/DatabricksInputPlugin.java (hunk 1 above)
Comment (yu-kioo) [reply src/main/java/org/embulk/input/DatabricksInputPlugin.java]: We have decided to always overwrite the UserAgentEntry value in options by user-agent, so please ignore the above comment. I removed condition as you commented.

**Review by chikamura — APPROVED** [review]
LGTM

**Pull request merged**